cmd/snaplock/runinhibit/inhibit.go

```diff
@@ -110,36 +185,67 @@ func Unlock(snapName string) error {
 	}
 	defer flock.Close()
 
+	// The following order of execution is important to avoid race conditions.
+	// Take the lock
 	if err := flock.Lock(); err != nil {
 		return err
 	}
+	// Write HintNotInhibited
 	f := flock.File()
-	return f.Truncate(0)
+	if err := f.Truncate(0); err != nil {
+		return err
+	}
+	if err := f.Sync(); err != nil {
+		return err
+	}
+	// Remove inhibit info file
+	if err := removeInhibitInfoFiles(snapName); err != nil {
+		return err
+	}
+
+	return nil
 }
 
 // IsLocked returns the state of the run inhibition lock for the given snap.
 //
 // It returns the current, non-empty hint if inhibition is in place. Otherwise
 // it returns an empty hint.
-func IsLocked(snapName string) (Hint, error) {
-	flock, err := osutil.OpenExistingLockForReading(HintFile(snapName))
+func IsLocked(snapName string) (Hint, InhibitInfo, error) {
```

Comment:
(very nitpick) This function `IsLocked` appears to be the only way callers have to read the inhibit info on disk. This makes sense -- there is no info unless the snap is locked -- but I wish it were clearer that this is how to read the contents. I think there was at least one case in non-test code where `IsLocked` was used only to read the the inhibit info, and not to check whether the snap was locked (though this must have been assumed).

Reply:
Great point, as you explained this came from the assumption that there is no info unless the snap is locked. I am with you that we might benefit from a clearer name but I can't think of something else.

@pedronis what do you think?

Reply:
I don't think i have brilliant suggestions, I suppose we could make it more similar to the lock function LockWithHint and call it IsLockedWithHint?  both have a bit a problem that one wonders wether one can lock without hint

Reply:
`IsLockedWithHint` gives me the impression that it is comparing existing hint with a given hint, similar to `waitInhibitUnlock`.